### ISSUE TITLE
swc-linux-arm64-gnu is not included in the GitHub releases after v1.7.14.

### ISSUE THREAD

**Issue by pan93412** [swc-project/swc#9769]
### Describe the bug

https://github.com/swc-project/swc/commit/f625035f8a21eb6d2bc487669a534257f3ef7c7c did not fix it.

---

The filename is swc-linux-arm64-gnu.

It was present not after version v1.7.14: [v1.7.14 (release)](https://github.com/swc-project/swc/releases/tag/v1.7.14). It has now disappeared. I suspect there was a change in the CI, but I cannot find the code that introduced this issue.

Discord discussion: [discord.com/channels/889779439272075314/889785438938726441/1290999698328649738](https://discord.com/channels/889779439272075314/889785438938726441/1290999698328649738).

### Input code

```typescript
N/A
```

### Config

```json
N/A
```

### Playground link (or link to the minimal reproduction)

https://play.swc.rs/?version=1.9.3&code=H4sIAAAAAAAAA0vOzyvOz0nVy8lP11DKSM3JyVfSBAAewMmrFAAAAA%3D%3D&config=H4sIAAAAAAAAA1WPSw7DIAwF9zkF8rrbdtE79BAWdSIifrKJVBTl7iUE0maH3xsz8jooBbNoeKq1PMsQkYX4nEsi2Sf8lARIOxTNJia49XaWvRrRCtVoOxpIyBOluiX3hoMNQajjLXPGmzH%2FC3VwkUnkCu4o%2BsnSVTc0JbjwXmrZDkk50qF%2FwA%2FqsvNjMPLqm4kXGrYvhlQioBQBAAA%3D

### SWC Info output

N/A

### Expected behavior

N/A

### Version

1.9.3

**Comment by kdy1** [comment]
cc @Brooooooklyn Is `linux-arm64-gnu` valid napi target?

**Comment by Brooooooklyn** [comment]
@pan93412 you can use `swc-linux-arm64-musl`

**Comment by pan93412** [comment]
> @pan93412 you can use `swc-linux-arm64-musl`

Yes, it works. However, there’s a library that downloads the SWC binary based on the runtime, and the missing of “glibc” prevents the library from downloading the latest SWC binaries. 😢 

I may create a PR for that library to ensure it always downloads the musl version if napi-rs would not generate the glibc version anymore.

**Comment by kdy1** [comment]
It's a bug of npm, not SWC

**Comment by lxjyo** [comment]
I have the similar problem. My version is 1.10.4, platform is darwin and arm64. It throw "Failed to load native binding".

**Comment by pan93412** [comment]
> I have the similar problem. My version is 1.10.4, platform is darwin and arm64. It throw "Failed to load native binding".

It might not be the same problem. You might need to open another issue to track it well.

**Comment by tbarrella** [comment]
Hello, I was wondering if I could help with this issue?

It looks like there are already builds for aarch64-unknown-linux-gnu, which I think is the same architecture as linux-arm64-gnu: https://github.com/swc-project/swc/blob/f33b0bc2d1474232f608847a6a26eaa068f2c106/.github/workflows/publish-npm-package.yml#L143

I'm not sure yet how publishing works; maybe a change is needed in one or both of these places to accommodate for the naming difference? and/or creating new directories in packages/ with arm64 -> aarch64?
- https://github.com/swc-project/swc/blob/main/packages/core/scripts/cli_artifacts.sh
- https://github.com/swc-project/swc/blob/main/scripts/cli_upload_gh_release.sh

**Comment by guw** [comment]
Some findings from the actions/workflow log.

[Publish 1.11.15](https://github.com/swc-project/swc/actions/runs/14187712145)
[Publish 1.11.15-nightly](https://github.com/swc-project/swc/actions/runs/14187712145/job/39747289649)

Output of "List binaries":

``` 
./packages/core/artifacts/bindings-core-1.11.15-nightly-20250401.1-aarch64-unknown-linux-gnu:
swc.linux-arm64-gnu.node

./packages/core/artifacts/bindings-core-1.11.15-nightly-20250[40](https://github.com/swc-project/swc/actions/runs/14187712145/job/39747289649#step:7:41)1.1-aarch64-unknown-linux-musl:
swc
swc.linux-arm64-musl.node
```

For `aarch64-unknown-linux-gnu` the `swc` binary is missing. This implies that the `cp` command was not successful either because the output does not exist or it was not run.


Scrolling through the "Build core ..." logs I noticed a possible issue:

```
115:  sh: 1: [[: not found
```

This leads me to believe that the `cp` is not executed because of the interesting command line. The last command executed seems to be `rustup target add aarch64-unknown-linux-gnu` and then it continues with `corepack`, skipping everything in between.

An indeed, the syntax is not supported in `sh`, which is the default shell of the `docker-run-action` action. See [their doc](https://github.com/swc-project/swc/pull/10303) for using bash.

**Comment by kdy1** [comment]
https://github.com/swc-project/swc/actions/runs/14194792959

I did, thanks!
